src/tiledb/cloud/bioimg/ingestion.py

```diff
@@ -48,13 +49,14 @@ def ingest(
     :param taskgraph_name: Optional name for taskgraph, defaults to None
     :param num_batches: Number of graph nodes to spawn.
         Performs it sequentially if default, defaults to 1
-    :param threads: Number of threads for node side multiprocessing, defaults to 8
+    :param threads: Number of threads for node side multiprocessing, defaults to 0
```

Comment:
Should this be 1?

Reply:
(also, I wish we had called this `processes` rather than `threads`, but alas - deprecate and replace in the future)

Reply:
The `threads` value is given to the `max_workers` argument in `ingestion_udf`. The reason for this rename is that DAG API has its own `max_workers` argument and I had to separate them without breaking the API. Successively this is passed to https://github.com/TileDB-Inc/TileDB-BioImaging/blob/f8365af24eb00714f851fef4a5b29993df7283b0/tiledb/bioimg/converters/base.py#L299. The default value there is `0`, thus is why kept it `0` in the upper level. When `0` we handle it as `None` inside the `bioimg.to_tiledb` https://github.com/TileDB-Inc/TileDB-BioImaging/blob/f8365af24eb00714f851fef4a5b29993df7283b0/tiledb/bioimg/converters/base.py#L589 so it will default to the number of processors on the machine. I called it `threads` cause we are using a `ThreadpoolExecutor`. I am open to renaming though I just tried to be consistent without breaking the API.

Reply:
@ktsitsi Would it be useful to follow https://docs.python.org/3/library/concurrent.futures.html#concurrent.futures.ThreadPoolExecutor as an example of paramererizing the number of workers? There the number of workers parameter defaults to None and then the documentation explains that the default is the number of processors * 5.

In our docstring above "defaults to 0" has the potential to mislead a user.